Python/modsupport.c

```diff
@@ -696,11 +710,5 @@ PyModule_AddType(PyObject *module, PyTypeObject *type)
     const char *name = _PyType_Name(type);
     assert(name != NULL);
 
-    Py_INCREF(type);
-    if (PyModule_AddObject(module, name, (PyObject *)type) < 0) {
-        Py_DECREF(type);
-        return -1;
-    }
-
-    return 0;
+    return PyModule_AddObjectRef(module, name, (PyObject *)type);
```

Comment:
What's the reason for not decref'ing here, as opposed to `PyModule_AddStringConstant` and `PyModule_AddIntConstant`? `PyModule_AddType` used to steal a reference, but now it doesn't, no?

Reply:
The caller holds a strong reference to type.

Reference counting is hard to get right. That's why I added a new function. Previously, Py_INCREF(type) was needed before calling PyModule_AddObject(). I let you count +1 and -1 on the ref count in all code paths and check manually if my PR doesn't anything (hint: it should not ;-)).

Reply:
> The caller holds a strong reference to type.
> 
> Reference counting is hard to get right. That's why I added a new function. Previously, Py_INCREF(type) was needed before calling PyModule_AddObject(). I let you count +1 and -1 on the ref count in all code paths and check manually if my PR doesn't anything (hint: it should not ;-)).

Argh, I though I had it right :) Thanks for the explanation! 🙏🏻